### ISSUE TITLE
Implement Local Hub integration tests as separate job

### ISSUE THREAD

**Issue by pkosiec** [capactio/capact#658]
## Description

Implement separate Hub integration tests:

- Extend tests for Hub
    - Move existing Local Hub tests out of our E2E integration tests suite https://github.com/capactio/capact/blob/955dd17c33e37db881257194d77c6fb4751b5e50/test/e2e/hub_test.go
    - Write additional tests for edge cases
    - Use Go, and our Go clients
- Local Hub:
  - Prepare test setup 
      - Neo4j + Hub JS + test delegated storage + run Go tests
      - We can use Docker compose
      - Write simple Go tests with assertions/requires from testify instead of using Gingko
  - Prepare list of tests for edge cases we want to test
      - Move and refactor tests from @mszostok's PRs
         - https://github.com/capactio/capact/pull/662/files#diff-84a321ed9a64e271345f3df51e1b24d27bff811bf63fb6a2446b3a7d9dcea5ae
         - https://github.com/capactio/capact/blob/9cd79fc829959b66172ebb9427ae0e01dbea4594/pkg/hub/client/local/client_test.go
      - Make sure all test cases are run for both TypeInstances that use built-in storage, and these that use delegated storage 
- Modify GitHub Actions pipeline and run these tests in parallel to our full E2E integration tests
- Test query for both typeinstance create value as string and value as object

This task is timebox for 3MD.

## Reason

Currently, we have a very basic happy path tests for our Public and Local Hub:
https://github.com/capactio/capact/blob/955dd17c33e37db881257194d77c6fb4751b5e50/test/e2e/hub_test.go

Unfortunately they don't test many edge cases (e.g. for Local Hub, deleting locked TypeInstance with the same ownerID). If we want to add more test cases, it would be very slow. That's why we need to split them to dedicated jobs, running without Kubernetes and with minimal set of Capact components, and then we can add another test cases.

**Comment by mkuziemko** [comment]
PR with creating the tests environment using docker-compose and moves tests from e2e was created: https://github.com/capactio/capact/pull/683.

Adding new test cases will be done in a new PR.
